account-kit/react/src/experimental/hooks/usePrepareCalls.ts

```diff
@@ -0,0 +1,94 @@
+import type { GetSmartWalletClientResult } from "@account-kit/core/experimental";
+import { type Address } from "viem";
+import type { SmartWalletClient } from "@account-kit/wallet-client";
+import {
+  type UseMutateAsyncFunction,
+  type UseMutateFunction,
+  useMutation,
+} from "@tanstack/react-query";
+import { clientHeaderTrack } from "@aa-sdk/core";
+import { useAccount as wagmi_useAccount } from "wagmi";
+import {
+  ClientUndefinedHookError,
+  UnsupportedEOAActionError,
+} from "../../errors.js";
+import { useAlchemyAccountContext } from "../../hooks/useAlchemyAccountContext.js";
+import { ReactLogger } from "../../metrics.js";
+
+export type UsePrepareCallsParams = {
+  client?: GetSmartWalletClientResult<Address>;
+};
+
+type MutationParams = Parameters<SmartWalletClient<Address>["prepareCalls"]>[0];
+
+type MutationResult = Awaited<
+  ReturnType<SmartWalletClient<Address>["prepareCalls"]>
+>;
+
+export type UsePrepareCallsResult = {
+  prepareCalls: UseMutateFunction<
+    MutationResult,
+    Error,
+    MutationParams,
+    unknown
+  >;
+  prepareCallsAsync: UseMutateAsyncFunction<
+    MutationResult,
+    Error,
+    MutationParams,
+    unknown
+  >;
+  preparedCalls: MutationResult | undefined;
+  isPreparingCalls: boolean;
+  error: Error | null;
+};
+
+export function usePrepareCalls(
```

Comment:
are these being exported from anywhere? if so they need jsdoc comments on them

Reply:
adding jsdoc now, but i don't think we are exporting any of the experimental hooks yet, so maybe we can do that all at once after they get merged into your base branch.